### PR TITLE
refactor: make param models defined as interface

### PR DESCRIPTION
**Pull request by eldenpark**
- AppError is renamed as `ApiError`, ApiResult became `ApiResponse`.
- For now, param models are defined as interface of TypeScript.
- error handler sends out ApiResponse, not some specially crafted object.

Fixes https://github.com/marmoym/marmoym/issues/135

### CODE REVIEW

File: src/middlewares/errorHandler.ts [hunk 1]
```diff
@@ -1,7 +1,8 @@
 import chalk from 'chalk';
 import { format } from 'util';
 
-import AppError from '@@models/AppError';
+import ApiResponse from '@@models/ApiResponse';
+import ApiError from '@@models/ApiError';
```
Comment (UnoRich) [inline src/middlewares/errorHandler.ts:4]: order

File: src/models/ResponseType.ts [hunk 2]
```diff
@@ -4,6 +4,11 @@ const ResponseType: ResponseType = {
     desc: 'email already used',
     label: 'EMAIL_ALREADY_USED',
   },
+  ERROR_WHILE_PROCESSING_ERROR: {
+    code: 501001,
+    desc: 'email already used',
```
Comment (gazzua) [inline src/models/ResponseType.ts]: change desc

File: src/routes/routers.ts [hunk 3]
```diff
@@ -7,9 +7,9 @@ import {
 } from 'express';
 import { createRouter } from '@nodekit/express-route-mapper';
 
-import ApiResult from '@@models/ApiResult';
+import ApiResponse from '@@models/ApiResponse';
```
Comment (UnoRich) [inline src/routes/routers.ts]: order

**Review by UnoRich — CHANGES_REQUESTED** [review]

File: src/models/ResponseType.ts [hunk 4]
```diff
@@ -34,6 +39,11 @@ const ResponseType: ResponseType = {
     desc: 'Request is routed but did not get the response',
     label: 'RESPONSE_NOT_PROVIDED',
   },
+  RESPONSE_TYPE_NOT_API_RESULT: {
+    code: 400001,
+    desc: 'Internal Error has occurred',
```
Comment (gazzua) [inline src/models/ResponseType.ts]: change desc

File: src/services/definitionService.ts [hunk 5]
```diff
@@ -1,13 +1,10 @@
...
-    return new ApiResult({});
-  },
-};
-
-export default definitionService;
+export async function addDefinition({
+  definition,
+  label,
+}: DefinitionAddParam): Promise<ApiResponse<any>> {
+  console.log(123, definition, label);
```
Comment (gazzua) [inline src/services/definitionService.ts]: delete console.log

**Review by gazzua — CHANGES_REQUESTED** [review]

**Review by gazzua — APPROVED** [review]

**Pull request merged**